packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/FixedEnumSerializationProvider.cs

```diff
@@ -116,16 +121,34 @@ protected override MethodProvider[] BuildMethods()
                 }
                 deserializationBody.Add(new IfStatement(condition)
                     {
-                        Return(new MemberExpression(_enumType.Type, enumField.Name))
+                        Return(new MemberExpression(EnumProvider.Type, enumField.Name))
                     });
             }
 
             // add a fallback throw statement to ensure every path of this method returns a value
-            deserializationBody.Add(Throw(New.ArgumentOutOfRangeException(_enumType, deserializationValueParameter)));
+            deserializationBody.Add(Throw(New.ArgumentOutOfRangeException(EnumProvider, deserializationValueParameter)));
 
             methods.Add(new(deserializationSignature, deserializationBody, this));
 
             return methods.ToArray();
         }
+
+        public ValueExpression ToSerial(ValueExpression enumExpression)
+        {
+            if (_enumType.ValueType.Equals(typeof(int)) || _enumType.ValueType.Equals(typeof(long)))
+            {
+                // when the fixed enum is implemented as int, we cast to the value
+                return enumExpression.CastTo(EnumProvider.EnumUnderlyingType);
+            }
+
+            // otherwise we call the corresponding extension method to convert the value
+            CSharpType? serializationType = SerializationProviders.FirstOrDefault()?.Type;
+            return enumExpression.Invoke($"ToSerial{_enumType.ValueType.Name}");
+        }
+
+        public ValueExpression ToEnum(ValueExpression valueExpression)
```

Comment:
I don't think we need these here anymore, there are snippets on CSharpTypeSnippet that provide this for us which we use in the MRWSerializationProvider

Same with ToSerial